### PR TITLE
Alerting: Do not retry rule evaluations with "input data must be a wide series but got type long" style errors

### PR DESCRIPTION
**Pull request by alexweav**
**What is this feature?**

Usually, this error happens with SQL or tabular datasources that return a dataframe that can't be converted to an Alerting-style series.

Therefore, it's in the same vein of problem as alerting's `invalidEvalResultFormatError` which we already consider as non retryable. This one just happens earlier in the stack. Consider it non-retryable as well.

**Why do we need this feature?**

Prevents unnecessary retries on an error that usually indicates that the query needs to be fixed.

**Who is this feature for?**

Users with this configuration problem who are using alerting evaluation retries - send fewer unnecessary queries.

**Which issue(s) does this PR fix?**:

n/a

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by JacobsonMT — APPROVED** [review]
LGTM :rocket: 

Comments from discussion outside of PR that are relevant but not blockers:

We can consider handing `ConversionError` wholesale instead of picking and choosing specific ones. The code does seem to imply this error is focused on frame structure/type compatibility. My understanding is that this is likely to be query-related and if not, unlikely to change after a fast retry. Though since we're not the core maintainers of this part, we won't be able to guarantee it stays this way even if it is this way now.

Basically, we'll have to decide between ensuring we cover all possible errors coming from `Convert`, or lean toward one of restrictive/lenient/retry semantics. Currently we've leniently retrying.

**Review by yuri-tceretian — APPROVED** [review]
LGTM
